flow/probes/ovssflow.go

```diff
@@ -173,50 +176,41 @@ func (o *OvsSFlowProbesHandler) UnregisterSFlowProbeFromBridge(bridgeUUID string
 	return nil
 }
 
-// RegisterProbeOnBridge registers a new probe on the OVS bridge
-func (o *OvsSFlowProbesHandler) RegisterProbeOnBridge(bridgeUUID string, tid string, capture *types.Capture, n *graph.Node) error {
+func (o *OvsSFlowProbesHandler) registerProbeOnBridge(bridgeUUID string, tid string, capture *types.Capture, n *graph.Node) error {
 	headerSize := flow.DefaultCaptureLength
 	if capture.HeaderSize != 0 {
 		headerSize = uint32(capture.HeaderSize)
 	}
 
-	opts := tableOptsFromCapture(capture)
-	ft := o.fpta.Alloc(tid, opts)
-
 	probe := OvsSFlowProbe{
 		ID:         bridgeUUID,
 		Interface:  "lo",
 		HeaderSize: headerSize,
 		Sampling:   1,
 		Polling:    10,
-		flowTable:  ft,
-	}
-
-	address := config.GetString("sflow.bind_address")
-	if address == "" {
-		address = "127.0.0.1"
 	}
 
-	addr := common.ServiceAddress{Addr: address, Port: 0}
-	agent, err := o.allocator.Alloc(bridgeUUID, probe.flowTable, capture.BPFFilter, headerSize, &addr, n, o.Graph)
-	if err != nil && err != sflow.ErrAgentAlreadyAllocated {
-		return err
-	}
+	if capture.Target == "" {
+		opts := tableOptsFromCapture(capture)
+		probe.flowTable = o.fpta.Alloc(tid, opts)
 
-	probe.Target = agent.GetTarget()
+		address := config.GetString("sflow.bind_address")
+		if address == "" {
```

Comment:
should default not be handled in `config/config.go` like most other parameters?

Reply:
not really, if the user provide an empty value in the config file default won't be applied